### ISSUE TITLE
Refactor: Open/Close events should be publicly exposed for top-level components

### ISSUE THREAD

**Issue by Elijbet** [Esri/calcite-design-system#4544]
**monday.com sync:** #18048591281

### Description

**Related Issue:** #2256

## Summary

Follow-up issue: Open/Close events should be publicly exposed, as these might be useful for users at some point. Especially for a top-level component that isn't emitting this event for another Calcite component to listen for. In that case, just ensure it does not have the @internal label or Internal prefix and is documented.

### Proposed Advantages

Users might need control over these.

### Which Component

Based on the comment below:
The originating issue for this has been addressed (making combobox open/close events public). Other components could implement OpenCloseComponent with associated events:

`open` prop:

- [x] action-menu (https://github.com/Esri/calcite-design-system/issues/6017)
- [x] block-section
- [x] list-item
- [x] menu-item

`expanded` prop:

- [x] accordion-item
- [x] action-bar
- [x] action-group
- [x] action-pad
- [x] tree-item

`closed` prop:

- [ ] list-item
- [ ] chip
- [ ] tab-title
- [ ] flow-item
- [ ] panel

`collapsed` prop:

- [x] panel
- [x] shell-panel
- [x] flow-item

We may also need to tackle https://github.com/Esri/calcite-design-system/issues/5074 along with this issue.

**Comment by jcfranco** [comment]
It seems the originating issue for this has been addressed (making combobox open/close events public). 

There are other components that could implement `OpenCloseComponent` with associated events:

**`open` prop**

* `action-menu` (https://github.com/Esri/calcite-design-system/issues/6017)
* `block-section`
* `list-item`
* `menu-item`

**`expanded` prop**

* `accordion-item`
* `action-bar`
* `action-group`
* `action-pad`
* `tree-item` (`expanded` prop)

Not sure if we want to address those in a separate PR or repurpose this one. 

We may also need to tackle https://github.com/Esri/calcite-design-system/issues/5074 along with this issue.

**Comment by driskull** [comment]
Do we have an issue to add these kind of events for `closed` props on `panel`, `flow-item`, `list-item`, etc?

**Comment by Elijbet** [comment]
> Do we have an issue to add these kind of events for `closed` props on `panel`, `flow-item`, `list-item`, etc?

I'll add those to this issue as well.

- `list-item`
- `panel` 
- `flow-item`
- `chip` 
- `tab-title`
- `tip` 
- `tip-manager`

**Comment by Elijbet** [comment]
In order to make components with expanded, collapsed, and closed props we will need to first address this issue: [Revisit event names emitted by openCloseComponent util](https://github.com/Esri/calcite-design-system/issues/9513#top)
#9513

**Comment by geospatialem** [comment]
Blocked by the efforts of #9513

**Comment by Elijbet** [comment]
The remaining components in the `closed` prop section already have publicly exposed `close` events, but are missing `open`, `beforeOpen`, and `beforeClose`.
